client/me/account/index.jsx

```diff
@@ -405,7 +457,7 @@ module.exports = React.createClass( {
 
 				<FormButton
 					isSubmitting={ this.state.submittingForm }
-					disabled={ ! this.props.userSettings.hasUnsavedSettings() || this.getDisabledState() }
+					disabled={ ! this.props.userSettings.hasUnsavedSettings() || this.getDisabledState() || !! this.hadFormError() }
```

Comment:
Ahh, I see we do the double bang here to cast to boolean. It may make more sense to do it in `hadFormError()` since that method name sounds like it would return a `boolean`.